velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp

```diff
@@ -242,47 +244,49 @@ struct MinMaxByNStringViewAccumulator {
 
   void
   compareAndAdd(C comparison, std::optional<V> value, Compare& comparator) {
-    if (base.topPairs.size() < base.n) {
-      addToAccumulator(comparison, value);
+    if (base.heapValues.size() < base.n) {
+      addToAccumulator(comparison, value, comparator);
     } else {
-      const auto& topPair = base.topPairs.top();
+      const auto& topPair = base.heapValues.front();
       if (comparator.compare(comparison, topPair)) {
-        freePair(topPair);
-        base.topPairs.pop();
-        addToAccumulator(comparison, value);
+        std::pop_heap(
+            base.heapValues.begin(), base.heapValues.end(), comparator);
+        addToFullAccumulator(comparison, value, comparator);
```

Comment:
nit: maybe we can just call 

```
base.heapValues.pop_back()
addToAccumulator(comparison, value, comparator);
```
We can then get rid of `addToFullAccumulator()`
`pop_back()` does not modify capacity and only modifies the size, so should be equivalent to what you want to achieve in `addToFullAccumulator()`